src/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -256,6 +256,11 @@ public Task<ExtensionEntry> LoadExtensionAsync(IExtensionInfo extensionInfo)
 
         public async Task<IEnumerable<FeatureEntry>> LoadFeaturesAsync()
         {
+            // for testing
+            Parallel.ForEach(GetFeatures(),
```

Comment:
This is wrong. Parallel.ForEach will never wait for the result, this only accepts an Action, not a Task.
The correct solution is to use the code below.
But it will only be multi-threading of the `LoadFeatureAsync` actually does something async and long running.